languages/go/exercises/concept/maps/.meta/example.go

```diff
@@ -0,0 +1,61 @@
+package maps
+
+// units store the Dozen Store unit measurement
+var units = map[string]int{
+	"quarter_of_a_dozen": 3,
+	"half_of_a_dozen":    6,
+	"dozen":              12,
+	"small_gross":        120,
+	"gross":              144,
+	"great_gross":        1728,
+}
+
+// bill to store the item that customer wants to buy
+var bill = map[string]int{}
+
+// AddItem add item to customer bill
+func AddItem(item string, unit string) bool {
```

Comment:
In this case I would expect `AddItem` to be a method on a type. The type could be `type bill map[string]int`. That way we only need `methods` as prerequisite. `methods` though has `functions`, `pointers`, etc. as prerequisite.

The alternative is that the `bills` map is passed into the `AddItem` function and then returned as well. I'd prefer this approach as it doesn't need additional prerequisites and it is generally a good practice to pass a map into a function to be changed and returned. Signature would then be:

`func AddItem(bill map[string]int,  item string, unit string) (map[string]int, bool)`

Reply:
Note: to have `units` global is ok, as it is used as a static list and not edited.

Reply:
Note2: same for the other functions.

Reply:
Note3: Why this change: currently `bill` is global. Meaning there can never be multiple bills. This doesn't really have an application in the real world and it would be nice if this exercise is as close to the real world as an exercise can be. Additionally `globals` (in the sense of global state, not global static variables) are considered "bad practice" and we shouldn't promote it.

Reply:
> In this case I would expect AddItem to be a method on a type. The type could be type bill map[string]int
>  alternative is that the bills map is passed into the AddItem function and then returned as well

I think both approach are fine, but I like the idea of passing struct as an argument

> are considered "bad practice" and we shouldn't promote it.

I don't want to get into an argument in this manner, but I feel like this is very opinionated, but I get it though, it promotes less cohesion. If you could refer me to what you considered as a bad practices that would be awesome.

Reply:
> alternative is that the bills map is passed into the AddItem function and then returned as well

Just came trough my mind, if you did that, you're moving the state of the package outside the package, that means the caller can modify the state and that would make the function useless. I don't think that's a good approach though.

Reply:
> If you could refer me to what you considered as a bad practices that would be awesome.

Without trying to start a big argument here: The bill variable is a `global state` variable. It is private to the package but still global. If 2 parts of your program tried to fill a bill at roughly the same time they would mess each others bills up.

> Just came trough my mind, if you did that, you're moving the state of the package outside the package, that means the caller can modify the state and that would make the function useless.

In this case the state handling would be up to the caller and all we do here is offer helper functions to modify the state, not to control it. That is how e.g. `append` and `delete` work. A slice is given to them, they modify it and return it.

I agree, that a custom type (e.g. a `struct` which hides the `map[string]int` aways) and can be intialized multiple times would be ideal for a `bill` object, but then we introduce more prerequisites to this exercise. This is what I am trying to avoid.

Reply:
I'm sorry, I disagree with your arguments. But I'll implement your approach later